pkg/kubelet/kuberuntime/labels.go

```diff
@@ -105,6 +106,14 @@ func newContainerLabels(container *v1.Container, pod *v1.Pod, containerType kube
 	labels[types.KubernetesContainerNameLabel] = container.Name
 	if utilfeature.DefaultFeatureGate.Enabled(features.DebugContainers) {
 		labels[types.KubernetesContainerTypeLabel] = string(containerType)
+		// It might be useful to store the spec for every container, but for now only store spec for debug containers
+		if containerType == kubecontainer.ContainerTypeEphemeral {
+			if containerJSON, e := json.Marshal(container); e != nil {
```

Comment:
Does size matter here? If it does a base64 encoded protobuf will probably be more efficient.